cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -559,12 +561,12 @@ private String subQuery(String index,CQLTermNode node, Table correlation) throws
 
     boolean isTermConstant = !tableNamePattern.matcher(term).matches();
 
-    String myField = dbTable.getTableName() + ".id";
-    String targetField = index2sqlText(foreignTableJsonb, fkey.getFieldName());
+    String myField = !indexInTable ?  dbTable.getTableName() + "." + PK_COLUMN_NAME : "Cast ( " + index2sqlText(dbTable.getTableName() + ".jsonb",fkey.getFieldName()) + " as UUID)";
```

Comment:
Using `Cast ( x as UUID )` is more difficult to read than `x::UUID`, can you change it here and in the following lines?